soccer/MainWindow.cpp

```diff
@@ -718,14 +703,54 @@ void MainWindow::updateStatus() {
     RJ::Time curTime = RJ::now();
 
     // Determine if we are receiving packets from an external referee
-    bool haveExternalReferee =
-        (curTime - ps.lastRefereeTime) < RJ::Seconds(0.5);
+    bool haveExternalReferee = (curTime - ps.lastRefereeTime) < RJ::Seconds(1);
+
+    std::vector<int> validIds = _processor->state()->ourValidIds();
 
-    /*if (_autoExternalReferee && haveExternalReferee &&
-    !_ui.externalReferee->isChecked())
-    {
-        _ui.externalReferee->setChecked(true);
-    }*/
+    for (int i = 1; i < 17; i++) {
```

Comment:
Can you remove 17 from this loop? In Constants.hpp, we have Num_Shells which you can use (which is 16, so you'll need to do NumShells+1 or use i+1 everywhere and start i at 0).